src/Middleware/MaintenanceMiddleware.php

```diff
@@ -126,4 +134,27 @@ public function mode($mode = null, $config = [])
 
         return $this->_mode = $mode;
     }
+    
+    /**
+     * Checks the whitelist against the current session IP.
+     *
+     * @param $ip
+     * @return bool
+     */
+    public function isWhitelisted($ip)
+    {
+        if (is_null($ip)) {
+            return false;
+        }
+
+        foreach ($this->whitelist as $bypassIp) {
+            if ($ip === $bypassIp) {
+                Configure::write('Wrench.bypass', $ip);
```

Comment:
I am not sure this is needed.

Just testing the value of `Configure::read('Wrench.enabled')` should be enough to know at the Controller level if the user accessing the page has a whitelisted IP address : if the user can reach the view, it will be because he / she has been whitelisted.